pkg/detectors/boxoauth/boxoauth.go

```diff
@@ -120,13 +120,8 @@ func verifyMatch(ctx context.Context, client *http.Client, id string, secret str
 			if err != nil {
 				return false, nil, err
 			}
-			body := string(bodyBytes)
-
-			if strings.Contains(body, "unauthorized_client") {
-				return true, nil, nil
-			}
-
-			return false, nil, nil
+			verified := !strings.Contains(string(bodyBytes), "unauthorized_client")
+			return verified, nil, nil
```

Comment:
Have we tested this API behavior?

If no, I think this change flips the logic in the wrong direction.

Box’s OAuth API behaves like this (based on the comment added above):
- If the client ID/secret are valid, the response contains "unauthorized_client" (since we’re intentionally sending a malformed request).
- If the credentials are invalid, the response contains "invalid_client".

With the updated code, `unauthorized_client` would mark the credentials as invalid, and `invalid_client` would incorrectly mark them as valid.